internal/config/config.go

```diff
@@ -27,8 +33,18 @@ const (
 // }
 type Target struct {
 	Project       string
-	Regions       []string
 	LabelSelector string
+
+	// Either "managed" or "gke".
```

Comment:
IMO in the long run we need to abstract this more and do something like:

```
type KServiceProvider interface {
	Services(ls labelSelector) ([]runapi.Service,error)
}
```

and we'd have GKEProvider, CloudRunManagedProvider.

Then they'd have their own built-in filters (e.g. cluster name, or regions) given at the time of initialization.

Otherwise any platform we add will overcrowd this Target type.

Reply:
I have abstracted most things into a `knative` package, but the abstraction I currently have happens after the target had been set. For any other Knative provider, we might just need an endpoint to the cluster. Thought I'd would think more about how to abstract the target too.

Reply:
As we discussed, we also need to consider authentication and other things, so this will be more complex. Let's work on this on another PR though